crates/alpen-ee/sequencer/src/batch_builder/traits.rs

```diff
@@ -0,0 +1,54 @@
+//! Core trait that defines the types for a batching strategy.
+
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use strata_acct_types::Hash;
+
+use super::Accumulator;
+
+/// Core trait that defines the types for a batching strategy.
+///
+/// The `BatchPolicy` trait allows different batching strategies to be implemented
+/// by specifying the types for block data and accumulated values, along with
+/// the accumulation logic.
+pub trait BatchPolicy: Send + Sync + 'static {
+    /// Data collected per block, used for sealing decisions.
+    /// This is provided by [`BlockDataProvider`](super::BlockDataProvider).
+    type BlockData: Send + Sync + Clone;
+
+    /// Accumulated value across blocks (e.g., count, DA size).
+    /// Must implement `Default` for initialization and reset.
+    type AccumulatedValue: Default + Send + Sync + Debug;
+
+    /// Accumulate block data into the value.
+    ///
+    /// Called when a block is added to the accumulator.
+    fn accumulate(value: &mut Self::AccumulatedValue, data: &Self::BlockData);
+}
+
+/// Policy for deciding when to seal a batch.
+///
+/// Implementations define the threshold logic for determining when a batch
+/// should be sealed (e.g., by block count, DA size, or a combination).
+pub trait BatchSealingPolicy<P: BatchPolicy>: Send + Sync {
+    /// Check if adding a block would exceed the batch threshold.
+    ///
+    /// If this returns `true`, the current batch should be sealed before
+    /// adding this block. The block will then become the first block of
+    /// the next batch.
+    ///
+    /// This is called with the current accumulator state and the data for
+    /// the block about to be added.
+    fn would_exceed(&self, accumulator: &Accumulator<P>, block_data: &P::BlockData) -> bool;
```

Comment:
Can a more generic name for this be `should_seal`? `would_exceed` as a trait method sounds too restricted for some policy that's only based on a threshold.